src/python/pants/engine/target.py

```diff
@@ -821,27 +821,6 @@ class AllTargetsRequest:
 # -----------------------------------------------------------------------------------------------
 
 
-@memoized
-def maybe_warn_dependencies_as_copied_field(tgt_type: type[TargetGenerator]) -> None:
```

Comment:
@stuhood do we need to error in this case?

Reply:
Ah, yeah - that seems to make sense. Not sure why I converted warns to raises elsewhere and not here. I'll wait for your ack though @stuhood before thrashing.

Reply:
Yea, this should be an error: sorry that that wasn't clearer. Thanks for tackling this.

Reply:
Plenty clear, just PEBKAC.